great_expectations/checkpoint/checkpoint.py

```diff
@@ -140,6 +144,7 @@ def run(
         batch_request = substituted_runtime_config.get("batch_request")
         validations = substituted_runtime_config.get("validations") or []
 
+        assert validations is not None
```

Comment:
This is an invariant based on the above logic. Just a simple way to remove IDE warnings but not sure how the team feels about asserts.

Reply:
it is of type `Optional[list]` - I could do `isinstance(validations, list)` too.

I believe the issue is that `get()` on L145 implicitly returns `Any`

Reply:
Can we make `get_validations_with_batch_request_as_dict` return `list`?
Or `cast` `validations` as a list?

Reply:
```python
validations = substituted_runtime_config.get("validations", [])
```

Does this work?

Reply:
I think the latter works!